docs/userguide/dev/codingguidelines.md

```diff
@@ -1065,17 +1075,17 @@ Calendar::createInstance(TimeZone* zone, UErrorCode& errorCode) {
     LocalPointer<TimeZone> adoptedZone(zone);
     if(U_FAILURE(errorCode)) {
         // The adoptedZone destructor deletes the zone.
-        return NULL;
+        return nullptr;
     }
     // since the Locale isn't specified, use the default locale
-    LocalPointer<Calendar> c(new GregorianCalendar(zone, Locale::getDefault(), errorCode));
-    if(c.isNull()) {
-        errorCode = U_MEMORY_ALLOCATION_ERROR;
-        // The adoptedZone destructor deletes the zone. return NULL;
-    } else if(U_FAILURE(errorCode)) {
+    LocalPointer<Calendar> c(new GregorianCalendar(zone, Locale::getDefault(), errorCode),
+                             errorCode);    // LocalPointer will set a U_MEMORY_ALLOCATION_ERROR if
+                                            // new GregorianCalendar() returns nullptr.
+    if(U_FAILURE(errorCode)) {
```

Comment:
This does not look right.
- If the memory allocation failed, then the adoptedZone destructor deletes the zone -- good.
- However, if the constructor was called and it failed, then it should still have adopted the zone, and we should call adoptedZone.orphan(); -- otherwise we get a double delete.
- In addition, the constructor may be called (and adopt the zone) but may itself fail with an OOM error. This means that we still need to check for c.isNull() in order to figure out whether the constructor adopted the zone.

Right?

Something like
```
    LocalPointer<Calendar> c(your new version with the extra errorCode));
    if (c.isValid()) {
        // c adopted the zone.
        adoptedZone.orphan();
    }
    if(U_FAILURE(errorCode)) {
        // If c was constructed, then the c destructor deletes the Calendar,
        // and the Calendar destructor deletes the adopted zone.
        return nullptr;
    }
    return c.orphan();
```

Reply:
Yes. Sorry about that. Took your suggestions.